### PR TITLE
Auto-roller accounts do not get labeled/commented with needs tests

### PR DESCRIPTION
**Pull request by ricardoamador**
## Change Description
Dependabot and auto-roller account PRs are exempt from `needs tests` label and the "It looks like this pull request may not have tests." comments.  

## Issue
Fix https://github.com/flutter/flutter/issues/105180

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/test/request_handlers/github_webhook_test.dart [hunk 1]
```diff
@@ -704,6 +709,90 @@ void main() {
       )).called(1);
     });
 
+    group("Auto-roller accounts do not label Framework PR with test label or comment.", () {
+      var inputs = {
+        'skia-flutter-autoroll',
+        'dependabot',
+      };
+
+      for (var element in inputs) {
```
Comment (keyonghan) [inline app_dart/test/request_handlers/github_webhook_test.dart:718]: Ditto.

File: app_dart/test/request_handlers/github_webhook_test.dart [hunk 2]
```diff
@@ -704,6 +709,90 @@ void main() {
       )).called(1);
     });
 
+    group("Auto-roller accounts do not label Framework PR with test label or comment.", () {
+      var inputs = {
```
Comment (keyonghan) [inline app_dart/test/request_handlers/github_webhook_test.dart:713]: Let's use explicit variable types.

File: app_dart/test/request_handlers/github_webhook_test.dart [hunk 3]
```diff
@@ -1282,6 +1371,105 @@ void foo() {
       )).called(1);
     });
 
+    group("Auto-roller accounts do not label Engine PR with test label or comment.", () {
+      var inputs = {
```
Comment (keyonghan) [inline app_dart/test/request_handlers/github_webhook_test.dart:1375]: Ditto.

File: app_dart/test/request_handlers/github_webhook_test.dart (hunk 2 above)
Comment (ricardoamador) [reply app_dart/test/request_handlers/github_webhook_test.dart:713]: You got it.

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 4]
```diff
@@ -413,6 +421,11 @@ class GithubWebhook extends RequestHandler<Body> {
       await gitHubClient.issues.addLabelsToIssue(slug, pr.number!, labels.toList());
     }
 
+    // We do not need to add test labels if this is an auto roller author.
+    if (config.rollerAccounts.contains(pr.user!.login)) {
```
[the remark this answers is not in the document]
Comment (ricardoamador) [reply app_dart/lib/src/request_handlers/github_webhook.dart:425]: Isn't it? I am not sure I understand.

[the remark this answers is not in the document]
Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/github_webhook.dart:425]: You're right, for some reason the GitHub diff showed it as a hardcoded value

**Review by CaseyHillers — APPROVED** [review]
flutter-lgtm

**Comment by CaseyHillers** [comment]
FYI: You can put "Fix" before an issue link in the pull request body to close said issue when this PR is merged.

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Pull request merged**